src/dbnode/storage/database.go

```diff
@@ -826,6 +826,12 @@ func (d *db) QueryIDs(
 	}
 	defer sp.Finish()
 
+	// Check if exceeding query limits at very beginning of
+	// query path to abandon as early as possible.
+	if err := d.opts.IndexOptions().QueryStats().UpdateDocs(0); err != nil {
```

Comment:
Would be good to hide this behind a`LimitExceeded()` function here too

Reply:
Will do